sdk/appconfiguration/app-configuration/test/public/secretReference.spec.ts

```diff
@@ -165,7 +173,13 @@ describe("AppConfigurationClient - SecretReference", () => {
         0,
         "Unexpected number of SecretReferences seen after updating"
       );
-      await client.deleteConfigurationSetting({ key: secondSetting.key });
+      if (secondSetting.isReadOnly) {
```

Comment:
Line 127 - readOnly is false
Line 152 - readOnly is true for the secondSetting (but the secondSetting.readOnly is never updated to true)

Line 176 - When you check `secondSetting.readOnly` which is false., the setReadOnly is not executed inside the if check since we assume the readOnly is false. But Line 152 had set it to true, which has to be reverted.

Reply:
Why do we allow this type of behavior like in Line 152? We can have the `secondSetting.readOnly` be false but `readOnly` is true for the setting?

Reply:
Update the test as you like, to make it clearer.

I am only pointing out the mistake here. 😉

Reply:
I get what you're saying now. Thank you for pointing it out! We can make a check with a `getConfigurationSetting` call, but there's not a case in which the `readOnly` is false for the setting, so making one extra call seems unnecessary. I don't feel strongly on this though.

Reply:
No, instead keep track of what is going on with the readOnly state better (locally) and verify if the setReadOnly call is needed.

Reply:
The call is needed since the resource can't be delete otherwise. I'm wondering if there's a way to sync the readOnly state and the setting's readOnly so it is easier to keep track